docs/tsp/typeServerProtocol.ts

```diff
@@ -0,0 +1,383 @@
+
+        // Name of the type.
+        name: string;
+
+        // Flags specific to the category
+        categoryFlags: number;
+    }
+
+    export interface Member {
+        // The name of the member.
+        name: string;
+
+        // The type of the member.
+        type: Type;
+
+        // The type the member came from
+        classType: Type | undefined;
+    }
+
+    // Flags that are used for searching for members of a type.
+    export const enum MemberAccessFlags {
+        None = 0,
+        SkipInstanceMembers = 1 << 0, // Skip instance members when searching for members of a type.
+        SkipObjectBaseClass = 1 << 1, // Skip members from the `object` base class when searching for members of a type.
+    }
+
+    // Represents the category of a declaration in the type system.
+    // This is used to classify declarations such as variables, functions, classes, etc.
+    export const enum DeclarationCategory {
+        Intrinsic = 0, // An intrinsic is a built-in type or function that is part of the language itself, such as `int`, `str`, `len`, etc.
+        Variable = 1, // A variable is a named storage location that can hold a value.
+        Param = 2, // A parameter is a variable that is passed to a function or method.
+        TypeParam = 3, // A type parameter is a placeholder for a type that can be specified when the type is used, such as in generics.
+        TypeAlias = 4, // A type alias is a name that refers to another type, such as `List[int]` or `Dict[str, int]`.
+        Function = 5, // A function is a named block of code that can be called with arguments and returns a value.
+        Class = 6, // A class is a blueprint for creating objects that encapsulates data and behavior.
+        SpecialBuiltInClass = 7, // Special built-in types like 'Tuple', 'Generic',  'Protocol', 'Callable', 'Type', etc
+        Alias = 8, // An alias declaration, which is a reference to another declaration.
+    }
+
+    // Represents a symbol declaration in the type system.
+    // A declaration is a specific instance of a symbol in the source code, such as a variable, function, or class.
+    export interface Declaration {
+        // Unique identifier for the declaration within the session.
+        handle: string | number;
+
+        // Category of this symbol (function, variable, etc.).
+        category: DeclarationCategory;
+
+        // Parse node associated with the declaration
+        node?: Node;
+
+        // The dot-separated import name for the file that
+        // contains the declaration (may not be definitive
+        // because a source file can be accessed via different
+        // import names in some cases).
+        moduleName: string;
+    }
+
+    // Synthesized type information for a declaration that is not directly
+    // represented in the source code, but is derived from the declaration. Other languages
+    // may refer to this as an anonymous type.
+    export interface SynthesizedTypeInfo {
```

Comment:
This is pretty specific to pyright, but I don't see how to eliminate it from the interface.